dataset/src/main/scala/frameless/ops/MatchesAll.scala

```diff
@@ -0,0 +1,19 @@
+package frameless
+package ops
+
+import shapeless.HList
+import shapeless.ops.hlist.Filter
+
+import scala.annotation.implicitNotFound
+
+
+@implicitNotFound("HList of type '${L}' is NOT comprised of only elements of type '${U}'")
+trait MatchesAll[L <: HList, U]
+
+object MatchesAll {
+  implicit def matchesAll[L <: HList, U, Rest <: HList]
+  (implicit
+   filtered: Filter.Aux[L, U, Rest],
+   remainedSame: Rest =:= L
```

Comment:
`Filter.Aux[L, U].Out =:= L` means that selecting all elements of type `U` in `L` is identity (that is, all elements of `L` have type `U`)

`FilterNot.Aux[L, U].Out =:= HNil` means that removing all elements of type `U` is `L` is the empty list (that is, all elements of `L` have type `U`)

I'm I missing something?